src/cacheHttpClient.ts

```diff
@@ -118,17 +128,19 @@ export async function downloadCache(
 
 // Reserve Cache
 export async function reserveCache(key: string): Promise<number> {
-    const restClient = createRestClient();
+    const httpClient = createHttpClient();
 
     const reserveCacheRequest: ReserveCacheRequest = {
         key
     };
-    const response = await restClient.create<ReserveCacheResponse>(
-        "caches",
+    const additionalHeaders = {
+        "Content-Type": "application/json"
+    };
+    const response = await httpClient.postJson<any>(
```

Comment:
@bryanmacfarlane re http-client

2: Having postJson<T> take T and return T means we cant have a separate request/response body Type, I left this common the the PR as well, and a suggestion of how to update it, but heres the example of the use-case

Reply:
3: postJson should maybe automatically set `Content-Type": "application/json`. I am having to set it here manually, I think the typed-rest-client did that automatically, because this port was failing until I sent these headers